rest/taskana-rest-spring/src/main/java/pro/taskana/rest/TaskController.java

```diff
@@ -330,13 +332,65 @@ private TaskQuery applyFilterParams(TaskQuery taskQuery, MultiValueMap<String, S
       params.remove(POR_VALUE);
     }
 
+    if (params.containsKey(PLANNED)) {
+
+      String[] instants = extractCommaSeparatedFields(params.get(PLANNED));
+
+      TimeInterval[] timeIntervals = extractTimeIntervals(instants);
+
+      taskQuery.plannedWithin(timeIntervals);
+
+      params.remove(PLANNED);
+    }
+
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug("Exit from applyFilterParams(), returning {}", taskQuery);
     }
 
     return taskQuery;
   }
 
+  private TimeInterval[] extractTimeIntervals(String[] instants) {
+
+    List<TimeInterval> timeIntervalsList = new ArrayList<>();
+
+    for (int i = 0; i < instants.length - 1; i += 2) {
+
+      TimeInterval timeInterval = determineTimeInterval(instants, i);
+
+      if (timeInterval != null) {
+
+        timeIntervalsList.add(timeInterval);
+      }
+    }
+
+    TimeInterval[] timeIntervalArray = new TimeInterval[timeIntervalsList.size()];
+
+    TimeInterval[] timeIntervals = timeIntervalsList.toArray(timeIntervalArray);
+
+    return timeIntervals;
+  }
+
+  private TimeInterval determineTimeInterval(String[] instants, int i) {
+
+    if (!instants[i].equals("maxTime") && !instants[i + 1].equals("maxTime")) {
```

Comment:
please consider making maxTime some kind of constant. also is this somewhere documented within the API documentation? this is like a magic string that suddenly does not need to be a valid timestamp / instant object and still have a meaning. 
Personally i would just remove this special cases and find a way to give "null" or only 1 timestamp such a meaning.

Reply:
Done. Also, we now offer 2 methods to filter. If one time interval is sufficient, the parameters "plannedFrom/dueFrom" and "plannedTo/dueTo" can be used. If one of those is left out it is an open time interval (like you suggested). For the other option to use multiple time intervals as filter, the parameter "planned" can be used and given an even amount of instants. For open time intervals an "empty" string can be used.

Reply:
what happens if someone uses both combinations? -> plannedFrom and planned. Or plannedFrom, plannedTo and planned? 

What is the expected & current behaviour? We open a lot of edge cases that need to be defined and handled.

Reply:
There is a check for invalid parameter combos in the new commit.